packages/astro/vendor/vite/dist/node/chunks/dep-35df7f96.js

```diff
@@ -56982,8 +56982,10 @@ const devHtmlHook = async (html, { path: htmlPath, server, originalUrl }) => {
                     .join('');
                 // add HTML Proxy to Map
                 addToHTMLProxyCache(config, url, scriptModuleIndex, contents);
-                // inline js module. convert to src="proxy"
-                s.overwrite(node.loc.start.offset, node.loc.end.offset, `<script type="module" src="${config.base + url.slice(1)}?html-proxy&index=${scriptModuleIndex}.js"></script>`);
+                const modulePath = `${config.base + htmlPath.slice(1)}?html-proxy&index=${scriptModuleIndex}.js`;
+                // invalidate the module so the newly cached contents will be served
+                server === null || server === void 0 ? void 0 : server.moduleGraph.invalidateId(config.root + modulePath);
```

Comment:
It took me a minute to get this one.

With a loose suggestion:
```suggestion
                if (server != null) server.moduleGraph.invalidateId(config.root + modulePath);
```

Or, with a loose suggestion:
```suggestion
                if (server !== null && server !== undefined) server.moduleGraph.invalidateId(config.root + modulePath);
```

Reply:
I wrote this as `server?.moduleGraph.invalidate`.... the code you are seeing here is generated by the build. All of the code in `vendor/vite` isn't particularly readable for that reason.

Reply:
Oh! Totally then.